9/README.md

```diff
@@ -0,0 +1,227 @@
+the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
+details. You should have received a copy of the GNU General Public License along with this program; if not, 
+see <http://www.gnu.org/licenses>.
+
+This Specification is a [free and open standard](http://www.digistan.org/open-standard:definition) and is governed by 
+the Digital Standards Organization's [Consensus-Oriented Specification System](http://www.digistan.org/spec:1/COSS).
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", 
+and "OPTIONAL" in this document are to be interpreted as described in [RFC 2119](http://tools.ietf.org/html/rfc2119).
+
+## Tango Data Types Specification
+
+Data Types are specifying format and limits of pieces of information exchanged between Tango Controls components.   
+
+### Goals
+
+Data Types aims to standardise a way how different kind of information is entered, displayed and processed within
+a Tango Controls systems irrespectively to its encoding in a transport protocol or computers' CPU architecture.
+It also allows to distinguish different kind of numeric data, string data, enumerations or multi-values data. 
+
+### Use Cases
+
+There are many use cases for Data Types. Some of them are listed below:
+
+* To specify expected data format for attributes, commands and pipes.
+
+* To display in the same way a numeric values read from a devices when some of them are using big-endian encoding and 
+  other use little-endian encoding.
+
+* To limit information processed to a practical level by choosing a Data Type of certain precision.
+  For example, a value of Data Type DevShort typically provides less precise information and consumes less computing 
+  resources than a value of Data Type DevLong.
+
+### DataType
+
+* Any value of Properties, Attributes, Pipes' fields and Commands' input and output arguments SHALL have defined its 
+  DataType.
+  
+* DataType SHALL be as follows: 
+
+```abnf
+ DataType = DevVoid 
+
+ DataType =/ DevBoolean 
+ 
+ DataType =/ DevShort / DevLong / DevLong64
+ DataType =/ DevUChar / DevUShort / DevULong / DevULong64
+ 
+ DataType =/ DevFloat / DevDouble
+
+ DataType =/ DevString  
+ 
+ DataType =/ DevVarBooleanArray / DevVarDoubleArray / DevVarFloatArray
+ DataType =/ DevVarShortArray / DevVarLongArray / DevVarLong64Array / DevVarCharArray
+ DataType =/ DevVarStringArray
+ DataType =/ DevVarUShortArray / DevVarULongArray / DevVarULong64Array
+ 
+ DataType =/ DevEncoded / DevVarEncodedArray
+ 
+ DataType =/ DevVarLongStringArray / DevVarDoubleStringArray
+ 
+ DataType =/ DevState
+  
+``` 
+
+
+
+### Numeric
+
+#### DevBoolean
+
+DevBoolean is a data type which values represents logical state of true or false.
+
+* DevBoolean SHALL be defined as follows:
+
+```abnf
+DevBoolean = "DEVBOOLEAN" / "DevBoolean"
```

Comment:
I don't know whether this is relevant but in cppTango, in enum CmdArgType, the enum associated to DevBoolean type is named DEV_BOOLEAN

Reply:
Thanks, @bourtemb. I will reserach this more, because it applies to other types, too.

Reply:
Together with @mliszcz,  we did a research and it seems there are no 'protocol' defined names for types except "CamelCase" strings commonly used in documentation and GUI applications. 

The `DEVBOOLEAN` and `DEV_BOOLEAN` are labels in enums for java and cpp, respectively. Even more, JTango is using also `Tango_DEV_BOOLEAN`. 

It seems, the more important are long values assigned to types, which can be found from both enums and consts defined in [JTango](https://github.com/tango-controls/JTango/blob/14a6527e3ca67339e51a905ef6c55d773ed04cb7/common/src/main/java/fr/esrf/TangoDs/TangoConst.java#L268) and probably related to the index of  [CmdArgTypeName array](https://github.com/tango-controls/cppTango/blob/4d2ad6555fd6bc6e6603406ea83b6ed0ed797eb3/cppapi/server/tango_const.h.in#L1024 ).

So, I suggest to specify data types names as camel case strings only and add info about associated long values. @hardion, @Ingvord, @t-b, I would be glad having your opinion, too :-)

Reply:
Finally, I left only "DevCamelCase" names.